test-cypress/integration/setup/my-settings.spec.js

```diff
@@ -78,6 +76,58 @@ describe('My Settings', () => {
             .and('be.disabled');
     });
 
+    it('Test SPARQL settings change', function () {
+        cy.importFromUrl(repositoryId, 'https://www.w3.org/TR/owl-guide/wine.rdf', {});
+        // Verify SPARQL settings for current user:
+        //     Go to SPARQL screen
+        // Verify that:
+        //     -Expand over sameAs is on (icon is SPARQL editor)
+        // -Inference is on (icon is SPARQL editor)
+        // -Total results count is working (run a query that generates more than 1000 results and verify that more than 1000 results are returned"
+        cy.visit('/sparql');
+        waitUntilSparqlPageIsLoaded();
+        verifySameAsIsActive(true);
+        verifyInferenceIsActive(true);
+
+        cy.pasteQuery('select * where { ?s ?p ?o . } limit 1002');
+        cy.get('#wb-sparql-runQuery').click();
+        cy.executeQuery();
+
+        cy.verifyResultsPageLength(1000);
+        cy.verifyResultsMessage('1,000 of 1,002');
+
+        // Change SPARQL settings:
+        //     Turn off sameAs and Inference settings from My Settings menu.
+        //     Uncheck Count total results checkbox
+        visitSettingsView();
+        toggleSameAs();
+        toggleInference();
+
+        // Verify SPARQL settings for current user:
+        //     Go to SPARQL screen
+        // Verify that:
+        // -Expand over sameAs is off (icon is SPARQL editor)
+        // -Inference is off (icon is SPARQL editor)
+        // -Total results count is working (run a query that generates more than 1000 results and verify that exactly 1000 results are returned"
+        cy.visit('/sparql');
+        waitUntilSparqlPageIsLoaded();
+        verifySameAsIsActive(false);
+        verifyInferenceIsActive(false);
+
+        cy.pasteQuery('select * where { ?s ?p ?o . } limit 1002');
+        cy.get('#wb-sparql-runQuery').click();
+        cy.executeQuery();
+
+        cy.verifyResultsPageLength(1000);
+        cy.verifyResultsMessage('1,000 of 1,000');
```

Comment:
Not sure if this is expected to work actually. But we can use the same scenario as testing inference/sameAs in sparql spec. Using OWL-Horst (Optimized) rule set when creating the repository and same results counts expectations since the idea here is not to test inference itself.